org.eclipse.xtext.util/src/org/eclipse/xtext/util/ReplaceRegion.java

```diff
@@ -56,6 +58,25 @@ public void applyTo(StringBuilder builder) {
 		builder.replace(offset, getEndOffset(), text);
 	}
 	
+	@Override
+	public boolean equals(Object object) {
+		if (object instanceof ReplaceRegion) {
+			ReplaceRegion rr = (ReplaceRegion) object;
+
+			boolean equals = true;
+			equals = equals && Objects.equals(getOffset(), rr.getOffset());
```

Comment:
I'm not a big fan of Object.equals(..) for primitives. It causes boxing and is distracting for the reader.
How does the equals / hashcode methods look like when they are generated by Eclipse?